File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        string result; // using this as a stack
        
        for(int i=0; i<s.size(); i++){
            result.push_back(s[i]);
            if(result.size() >= part.size() && result.substr(result.size() - part.size()) == part){
                int n = part.size();
                while(n > 0){
                    result.pop_back();
                    n = n - 1;
                }
            }
        }
        return result;
    }
};

File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder result = new StringBuilder();
        for(int i=0; i<s.length(); i++){
            result.append(s.charAt(i));
            if(result.length() >= part.length() && result.substring(result.length() - part.length()).equals(part)){
                int n = part.length();
                while(n > 0){
                    result.deleteCharAt(result.length()-1);
                    n = n - 1;
                }
            }
        }
        return result.toString();
    }
}